### ISSUE TITLE
Failue to assign a std::variant

### ISSUE THREAD

**Issue by mu001999** [microsoft/vscode-cpptools#6623]
```cpp
#include <variant>

int main()
{
    std::variant<int, double> v;
    v = 1; // marks as error
    v = 1.2; // marks as error
}
```

I have set `cppStandard` to `c++17` and compiled it successfully. But I got the red underline with the error says:
> no operator "=" matches these operands -- operand types are: std::variant<int, double> = int

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. I've filed a bug on our shared VS parser at https://developercommunity.visualstudio.com/content/problem/1278119/c-intellisense-gives-errors-with-the-variant-heade.html .
 FYI, it only repros when clang is used (not gcc or cl.exe), changing your compilerPath to g++ may be a workaround.

**Comment by mu001999** [comment]
Thanks!

**Comment by mmatrosov** [comment]
See also #4636

**Comment by sean-mcmanus** [comment]
I created a new issue at https://developercommunity.visualstudio.com/content/problem/1331002/c-intellisense-gives-errors-with-the-gcc-102-varia.html

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0-insiders

**Comment by michelleangela** [comment]
Fix is released in the 1.3.0 release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0
